python/demo/poisson/demo_poisson.py

```diff
@@ -76,30 +76,37 @@
 import numpy as np
 
 import ufl
-from dolfinx import fem, plot
-from dolfinx.fem import FunctionSpace, dirichletbc, locate_dofs_topological
-from dolfinx.io import XDMFFile
-from dolfinx.mesh import CellType, create_rectangle, locate_entities_boundary
-from ufl import ds, dx, grad, inner
-
+from dolfinx import fem, plot, io, mesh as _mesh
 from mpi4py import MPI
 from petsc4py.PETSc import ScalarType
 
 # We begin by defining a mesh of the domain and a finite element
-# function space :math:`V` relative to this mesh. As the unit square is
-# a very standard domain, we can use a built-in mesh provided by the
-# class :py:class:`create_unit_square_mesh
-# <dolfinx.mesh.create_unit_square_mesh>`. In order to create a mesh
-# consisting of 32 x 32 squares with each square divided into two
-# triangles, we do as follows ::
-
-# Create mesh and define function space
-mesh = create_rectangle(MPI.COMM_WORLD, ((0.0, 0.0), (2.0, 1.0)), (32, 16), CellType.triangle)
-V = FunctionSpace(mesh, ("Lagrange", 1))
+# function space :math:`V` relative to this mesh. We create a
+# rectangular mesh using built-in function provided by
+# class :py:class:`create_rectangle<dolfinx.mesh.create_rectangle>`.
+# In order to create a mesh consisting of 32 x 16 squares with each
+# square divided into two triangles, we do as follows ::
+
+# Create an MPI communicator
+comm = MPI.COMM_WORLD
+
+# Create mesh
+mesh = _mesh.create_rectangle(
+    comm=comm,
+    points=((0.0, 0.0), (2.0, 1.0)),
+    n=(32, 16),
+    cell_type=_mesh.CellType.triangle
+)
+# Define element which is a tuple with (family, degree, form_degree).
+# We can also omit the form_degree and use the default value
+element = ("Lagrange", 1)
```

Comment:
I don't find the extra assignment here so helpful. I prefer
```
V = fem.FunctionSpace(mesh, ("Lagrange", 1))
```

Reply:
Fixed in https://github.com/FEniCS/dolfinx/pull/1979/commits/b48e86cbdefc812f07b0e2f131a7d799853168fe